src/app/app.component.ts

```diff
@@ -51,12 +51,26 @@ export class AppComponent implements OnInit {
     if (params.get('hideGoogle')) {
       this.globalVars.hideGoogle = true;
     }
-    
+
+    // Callback should only be used in mobile applications, where payload is passed through URL parameters.
+    if (params.get('callback')) {
+      try {
+        this.globalVars.callback = new URL(params.get('callback') as string);
+        this.globalVars.isCallbackValid = true;
```

Comment:
How is callback related to hostname? Do we want to enforce that the callback has the same hostname as the thing that called this? I don't think it's necessary but put some juices on it and lmk.

Reply:
@diamondhands0 I believe mobile clients have localhost hostname. To make callbacks safer, we had this idea to whitelist URL schemes for callbacks on a case by case basis, starting with e.g. `identity://`. I remember it's possible to have a custom, static scheme on iOS, but not sure if that's the case for Android. We should discuss more on Slack.

Reply:
What's safer about enforcing identity://? Seems equally safe/unsafe compared to allowing an arbitrary URL scheme doesn't it? Either way, I think this is a fine threat model as-is. Basically, if the user hits selectAccountAndDeriveKey, then they're at the mercy of whatever callback URL the caller set, which makes sense.

Resolving.